geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQueryService.java

```diff
@@ -104,10 +104,19 @@
    * implementation of MethodInvocationAuthorizer, which provides the same functionality.
    */
   @Deprecated
-  public static final boolean ALLOW_UNTRUSTED_METHOD_INVOCATION = Boolean.getBoolean(
-      GEMFIRE_PREFIX + "QueryService.allowUntrustedMethodInvocation");
-
-  private boolean deprecatedWarningHasBeenShown = false;
+  public static final boolean ALLOW_UNTRUSTED_METHOD_INVOCATION;
+
+  static {
+    String deprecatedValue =
+        System.getProperty(GEMFIRE_PREFIX + "QueryService.allowUntrustedMethodInvocation");
+    if (deprecatedValue != null) {
+      logger.warn("The property " + GEMFIRE_PREFIX +
+          "QueryService.allowUntrustedMethodInvocation is deprecated. To provide the same " +
+          "functionality, please use the UnrestrictedMethodAuthorizer implementation of" +
+          "MethodInvocationAuthorizer");
```

Comment:
There's a missing space between `of` and `MethodInvocationAuthorizer` here.
Also, and sorry for nitpicking here, I think it would be better if the message reads something like _X is deprecated, please use Y instead_.

Reply:
Done